.buildkite/bk.fips-integration.pipeline.yml

```diff
@@ -0,0 +1,169 @@
+    env:
+      ASDF_TERRAFORM_VERSION: 1.9.2
+    command: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+      source .buildkite/scripts/steps/ess_start.sh
+    artifact_paths:
+      - test_infra/ess/*.tfstate
+      - test_infra/ess/*.lock.hcl
+    agents:
+      image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci-with-hooks:0.5"
+      useCustomGlobalHooks: true
+
+  - group: "fips:Stateful:Ubuntu"
+    key: integration-tests-ubuntu-fips
+    depends_on:
+      - integration-fips-ess
+    steps:
+      - label: "fips:x86_64:non-sudo:{{matrix}}"
+        depends_on:
+          - packaging-ubuntu-x86-64-fips
+        env:
+          FIPS: "true"
+        command: |
+          buildkite-agent artifact download build/distributions/** . --step 'packaging-ubuntu-x86-64-fips'
+          .buildkite/scripts/steps/integration_tests_tf.sh {{matrix}} false
+        artifact_paths:
+          - build/**
+          - build/diagnostics/**
+        retry:
+          automatic:
+            limit: 1
+        agents:
+          provider: "aws"
+          image: "${IMAGE_UBUNTU_X86_64_FIPS}"
+          instanceType: "m5.2xlarge"
+        matrix:
+          - fleet # currently there is only a single test in the fleet group, add more tests once they have been defined
+
+      - label: "fips:x86_64:sudo:{{matrix}}"
+        depends_on:
+          - packaging-ubuntu-x86-64-fips
+        env:
+          FIPS: "true"
+        command: |
+          buildkite-agent artifact download build/distributions/** . --step packaging-ubuntu-x86-64-fips
+          .buildkite/scripts/steps/integration_tests_tf.sh {{matrix}} true
+        artifact_paths:
+          - build/**
+          - build/diagnostics/**
+        retry:
+          automatic:
+            limit: 1
+        agents:
+          provider: "aws"
+          image: "${IMAGE_UBUNTU_X86_64_FIPS}"
+          instanceType: "m5.2xlarge"
+        matrix:
+          - fleet
+
+      - label: "fips:arm64:non-sudo:{{matrix}}"
+        depends_on:
+          - packaging-ubuntu-arm64-fips
+        env:
+          FIPS: "true"
+        command: |
+          buildkite-agent artifact download build/distributions/** . --step 'packaging-ubuntu-arm64-fips'
+          .buildkite/scripts/steps/integration_tests_tf.sh {{matrix}} false
+        artifact_paths:
+          - build/**
+          - build/diagnostics/**
+        retry:
+          automatic:
+            limit: 1
+        agents:
+          provider: "aws"
+          image: "${IMAGE_UBUNTU_ARM64_FIPS}"
+          instanceType: "m6g.2xlarge"
+        matrix:
+          - fleet
+
+      - label: "fips:arm64:sudo:{{matrix}}"
+        depends_on:
+          - packaging-ubuntu-arm64-fips
+        env:
+          FIPS: "true"
+        command: |
+          buildkite-agent artifact download build/distributions/** . --step packaging-ubuntu-arm64-fips
+          .buildkite/scripts/steps/integration_tests_tf.sh {{matrix}} true
```

Comment:
What's the reason we don't use a matrix entry for the `true` and `false` values?, as far as I see `"fips:arm64:non-sudo:{{matrix}}"` and `"fips:arm64:sudo:{{matrix}}"` only differ in the second argument?

The same for the other one

Reply:
I copied the approach in `bk.integration.pipeline.yml`; i think it's so that the integration tests have increased parallel execution in that pipeline. I don't know if we need that here though, @ycombinator what do you think? I don't think we will have too many FIPS specific tests does it make sense to use less runners?

Reply:
I've changed to use multi-axis matrixes, I think we may only want to add a group or two for FIPS testing